### PR TITLE
chore(CIS): grub configuration changes to accommodate CIS

### LINKED ISSUE #1009: CIS 4.1.3 Ensure auditing for processes that start prior to auditd is enabled
From chef:

```
cookbook_file 'Set /etc/default/grub' do
  path '/etc/default/grub'
  source 'etc/default/grub'
end

execute 'Update grub2 configuration' do
  command 'update-grub'
end
```

### PR DESCRIPTION
**Pull request by jackfrancis**
**Reason for Change**:

Enables auditing via grub, see https://secscan.acron.pl/centos7/4/1/3

**Issue Fixed**:

Fixes #1009

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

**Comment by jackfrancis** [comment]
Validated w/ "ubuntu" distro:

```
  should validate all cloud-init-paved files
  /Users/jackfrancis/work/src/github.com/Azure/aks-engine/test/e2e/kubernetes/kubernetes_test.go:237

$ k config view -o json

$ k get nodes -o json

$ scp -i /Users/jackfrancis/work/src/github.com/Azure/aks-engine/_output/kubernetes-westus2-28812-ssh -o StrictHostKeyChecking=no scripts/cloud-init-files-validate.sh azureuser@kubernetes-westus2-28812.westus2.cloudapp.azure.com:/tmp/cloud-init-files-validate.sh
2019/04/22 10:06:42 Authorized uses only. All activity may be monitored and reported.


$ ssh -A -i /Users/jackfrancis/work/src/github.com/Azure/aks-engine/_output/kubernetes-westus2-28812-ssh -o ConnectTimeout=30 -o StrictHostKeyChecking=no azureuser@kubernetes-westus2-28812.westus2.cloudapp.azure.com -p 22 scp -o StrictHostKeyChecking=no /tmp/cloud-init-files-validate.sh k8s-agentpool1-12763148-vmss000000:/tmp/cloud-init-files-validate.sh

$ ssh -A -i /Users/jackfrancis/work/src/github.com/Azure/aks-engine/_output/kubernetes-westus2-28812-ssh -p 22 -o ConnectTimeout=10 -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o LogLevel=ERROR azureuser@kubernetes-westus2-28812.westus2.cloudapp.azure.com ssh -o ConnectTimeout=10 -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o LogLevel=ERROR k8s-agentpool1-12763148-vmss000000 "/tmp/cloud-init-files-validate.sh"

$ ssh -A -i /Users/jackfrancis/work/src/github.com/Azure/aks-engine/_output/kubernetes-westus2-28812-ssh -o ConnectTimeout=30 -o StrictHostKeyChecking=no azureuser@kubernetes-westus2-28812.westus2.cloudapp.azure.com -p 22 scp -o StrictHostKeyChecking=no /tmp/cloud-init-files-validate.sh k8s-agentpool1-12763148-vmss000001:/tmp/cloud-init-files-validate.sh

$ ssh -A -i /Users/jackfrancis/work/src/github.com/Azure/aks-engine/_output/kubernetes-westus2-28812-ssh -p 22 -o ConnectTimeout=10 -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o LogLevel=ERROR azureuser@kubernetes-westus2-28812.westus2.cloudapp.azure.com ssh -o ConnectTimeout=10 -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o LogLevel=ERROR k8s-agentpool1-12763148-vmss000001 "/tmp/cloud-init-files-validate.sh"

$ ssh -A -i /Users/jackfrancis/work/src/github.com/Azure/aks-engine/_output/kubernetes-westus2-28812-ssh -o ConnectTimeout=30 -o StrictHostKeyChecking=no azureuser@kubernetes-westus2-28812.westus2.cloudapp.azure.com -p 22 scp -o StrictHostKeyChecking=no /tmp/cloud-init-files-validate.sh k8s-master-12763148-0:/tmp/cloud-init-files-validate.sh

$ ssh -A -i /Users/jackfrancis/work/src/github.com/Azure/aks-engine/_output/kubernetes-westus2-28812-ssh -p 22 -o ConnectTimeout=10 -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o LogLevel=ERROR azureuser@kubernetes-westus2-28812.westus2.cloudapp.azure.com ssh -o ConnectTimeout=10 -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o LogLevel=ERROR k8s-master-12763148-0 "/tmp/cloud-init-files-validate.sh"
•
```

File: test/e2e/kubernetes/scripts/cloud-init-files-validate.sh [hunk 1]
```diff
@@ -15,4 +16,6 @@ done
 # verify that no files under /var/log have read access to everyone
 sudo find /var/log -type f -perm '/o+r' | (! grep ^) || exit 1
 # specific file mode validations
-stat /boot/grub/grub.cfg | grep 'Access: (0400' || exit 1
\ No newline at end of file
+stat /boot/grub/grub.cfg | grep 'Access: (0400' || exit 1
+# validate grub configuration
+sudo grep "^\s*linux" /boot/grub/grub.cfg | grep 'audit=1' || exit 1
```
Comment (mboersma) [inline test/e2e/kubernetes/scripts/cloud-init-files-validate.sh]: Should we also grep for `ipv6.disable=1` or is that not part of the CIS recommendation?

File: parts/k8s/cloud-init/artifacts/default-grub [hunk 2]
```diff
@@ -0,0 +1,35 @@
+# If you change this file, run 'update-grub' afterwards to update
```
Comment (CecileRobertMichon) [inline parts/k8s/cloud-init/artifacts/default-grub]: Can we make this file "lighter" by removing a lot of the commented out lines?

File: test/e2e/kubernetes/scripts/cloud-init-files-validate.sh (hunk 1 above)
Comment (jackfrancis) [reply test/e2e/kubernetes/scripts/cloud-init-files-validate.sh]: Yeah, in fact this grep is carried over specifically from here:

https://secscan.acron.pl/centos7/4/1/3

**Pull request merged**